diff_cover/diff_cover_tool.py

```diff
@@ -92,6 +94,13 @@ def parse_coverage_args(argv):
         help=MARKDOWN_REPORT_HELP,
     )
 
+    parser.add_argument(
+        "--github-warning-annotations",
```

Comment:
given the idea of moving to a plugin type framework, maybe this should be implemented as a top level `--format` flag, similar to some other programs, so that it's possible to do `--format github-native` or `--format xyz`?

There would have to be discussion then about whether multiple formats are allowed (and if so, how)... for example, would it allow `--format json,markdown,github-native` or multiple instances of the same flag, something like `--format github-native --format markdown:foo.md` (the latter case allowing specifying the format and filename together)? If it's too invasive to take over the markdown / json / etc. reports, could at least have a single flag to control all stdout / stderr reporting.

Reply:
Yep, I like this

Reply:
I will say if we move to this style then we should have it support the existing formats as well as the new one. 

We kinda have to keep the existing flags to avoid breaking compatibility, but if we are going to be adding more then moving to something like this makes sense

Reply:
I actually started down the path of attempting to pass something like `--github-annotations=warning` or `--github-annotations=error`.
But I saw that customizing the context that is passed to jinja template for specific template generators wasn't so easy to adapt.

Reply:
@timkrins if you wanted to have that context could that be achieved by adding some kind of `context` parameter to the __init__ of BaseReportGenerator?

Reply:
Maybe - I'll try have a go later in the week

Reply:
This has changed significantly, and adding `--format github` is now possible.